ykrt/src/compile/jitc_yk/opt/mod.rs

```diff
@@ -77,13 +78,15 @@ impl Opt {
         let mut iidx_map = vec![InstIdx::max(); base];
         let skipping = self.m.iter_skipping_insts().collect::<Vec<_>>();
         for (iidx, inst) in skipping.into_iter() {
-            let c = inst.dup_and_remap_vars(&mut self.m, |op_iidx: InstIdx| {
-                Operand::Var(iidx_map[usize::from(op_iidx)])
-            })?;
-            let copy_iidx = self.m.push(c)?;
-            iidx_map[usize::from(iidx)] = copy_iidx;
             match inst {
                 Inst::TraceHeaderStart => {
+                    self.m.trace_body_start = self.m.trace_header_start().to_vec();
```

Comment:
This has the disadvantage that we have to iterate over these values twice. Once when cloning them and again during remapping. Why don't we just `assert` here that we can only see `TraceHeaderStart` once per `Module`?

Reply:
I agree, we could probably fix that. The performance difference will be tiny and I'm not quite sure how to do it without adding more code.